### PR TITLE
Added test stubs for - user timezone and Dashboard UI

### PR DESCRIPTION
**Pull request by tkolhar**
please review it
thanks

$ flake8 test_user.py 
gives clean output

$ flake8 test_dashboard.py
gives clean output

### CODE REVIEW

File: tests/foreman/ui/test_dashboard.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+    @stubbed()
+    @tier3
+    def test_save_dashboard(self):
+        """@Test: Save the Dashboard UI
+
+        @Feature: Dashboard
+
+        @Steps:
+        1.Navigate to Monitor -> Content Dashboard
```
Comment (elyezer) [inline tests/foreman/ui/test_dashboard.py]: Add a blank line between steps tag and the steps list.

File: tests/foreman/ui/test_dashboard.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+    @stubbed()
+    @tier3
+    def test_reset_dashboard(self):
+        """@Test: Reset the Dashboard to default UI
+
+        @Feature: Dashboard
+
+        @Steps:
+        1.Navigate to Monitor -> Content Dashboard
```
Comment (elyezer) [inline tests/foreman/ui/test_dashboard.py]: Add a blank line between steps tag and the steps list.

File: tests/foreman/ui/test_dashboard.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+    @stubbed()
+    @tier3
+    def test_addwidgets_dashboard(self):
+        """@Test: Add Widgets to Dashboard UI
+
+        @Feature: Dashboard
+
+        @Steps:
+        1.Navigate to Monitor -> Content Dashboard
```
Comment (elyezer) [inline tests/foreman/ui/test_dashboard.py]: Add a blank line between steps tag and the steps list.

File: tests/foreman/ui/test_dashboard.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+
+    @stubbed()
+    @tier3
+    def test_bookmark_search_dashboard(self):
+        """@Test: Bookmark the search filter in Dashboard UI
+
+        @Feature: Dashboard
+
+        @Steps:
+        1.Navigate to Monitor -> Content Dashboard
```
Comment (elyezer) [inline tests/foreman/ui/test_dashboard.py]: Add a blank line between steps tag and the steps list.

File: tests/foreman/ui/test_dashboard.py [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    @stubbed()
+    @tier3
+    def test_host_configuration_status_dashboard(self):
+        """@Test: Check if Host Configuration Status
+        Widget links are working
+
+        @Feature: Dashboard
+
+        @Steps:
+        1.Navigate to Monitor -> Content Dashboard
```
Comment (elyezer) [inline tests/foreman/ui/test_dashboard.py]: Add a blank line between steps tag and the steps list.

File: tests/foreman/ui/test_dashboard.py [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+    @stubbed()
+    @tier3
+    def test_host_configuration_dashboard(self):
+        """@Test: Check if Host Configuration Chart
+        is working in Dashboard UI
+
+        @Feature: Dashboard
+
+        @Steps:
+        1.Navigate to Monitor -> Content Dashboard
```
Comment (elyezer) [inline tests/foreman/ui/test_dashboard.py]: Add a blank line between steps tag and the steps list.

File: tests/foreman/ui/test_dashboard.py [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+
+    @stubbed()
+    @tier3
+    def test_task_status_dashboard(self):
+        """@Test: Check if Task Status Dashboard UI
+
+        @Feature: Dashboard
+
+        @Steps:
+        1.Navigate to Monitor -> Content Dashboard
```
Comment (elyezer) [inline tests/foreman/ui/test_dashboard.py]: Add a blank line between steps tag and the steps list.

File: tests/foreman/ui/test_dashboard.py [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+    @stubbed()
+    @tier3
+    def test_warning_error_task_status_dashboard(self):
+        """@Test: Check if Latest Warning/Error
+        Tasks Status are working in Dashboard UI
+
+        @Feature: Dashboard
+
+        @Steps:
+        1.Navigate to Monitor -> Content Dashboard
```
Comment (elyezer) [inline tests/foreman/ui/test_dashboard.py]: Add a blank line between steps tag and the steps list.

File: tests/foreman/ui/test_dashboard.py [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+    @stubbed()
+    @tier3
+    def test_contentview_history_dashboard(self):
+        """@Test: Check if Content View History
+           are working in Dashboard UI
+
+        @Feature: Dashboard
+
+        @Steps:
+        1.Navigate to Monitor -> Content Dashboard
```
Comment (elyezer) [inline tests/foreman/ui/test_dashboard.py]: Add a blank line between steps tag and the steps list.

File: tests/foreman/ui/test_dashboard.py [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+    @stubbed()
+    @tier3
+    def test_discovered_host_dashboard(self):
+        """@Test: Check if user can access
+           Discovered Host [Widget]
+
+        @Feature: Dashboard
+
+        @Steps:
+        1.Navigate to Monitor -> Content Dashboard
```
Comment (elyezer) [inline tests/foreman/ui/test_dashboard.py]: Add a blank line between steps tag and the steps list.

File: tests/foreman/ui/test_user.py [hunk 11]
```diff
@@ -1027,3 +1027,74 @@ def test_negative_create_bookmark_with_blank_query(self):
...
+
+    @stubbed()
+    @tier3
+    def test_set_timezone(self):
+        """@Test: Set new timezone for a user
+
+        @Feature: User timezone
+
+        @Steps:
+        1.Navigate to Administer -> Users
```
Comment (elyezer) [inline tests/foreman/ui/test_user.py]: Add a blank line between steps tag and the steps list.

File: tests/foreman/ui/test_user.py [hunk 12]
```diff
@@ -1027,3 +1027,74 @@ def test_negative_create_bookmark_with_blank_query(self):
...
+
+    @stubbed()
+    @tier3
+    def test_dashboard_shows_newtime(self):
+        """@Test: Check if dashboard shows time according to the timezone set
+
+        @Feature: User timezone
+
+        @Steps:
+        1.Change the timezone for a user in Administer -> Users tab
```
Comment (elyezer) [inline tests/foreman/ui/test_user.py]: Add a blank line between steps tag and the steps list.

File: tests/foreman/ui/test_user.py [hunk 13]
```diff
@@ -1027,3 +1027,74 @@ def test_negative_create_bookmark_with_blank_query(self):
...
+
+    @stubbed()
+    @tier3
+    def test_logfiles_shows_newtime(self):
+        """@Test: Check if logfiles reflect the new timezone set by the user
+
+        @Feature: User timezone
+
+        @Steps:
+        1.Change the timezones for user in Administer -> Users Tab
```
Comment (elyezer) [inline tests/foreman/ui/test_user.py]: Add a blank line between steps tag and the steps list.

File: tests/foreman/ui/test_user.py [hunk 14]
```diff
@@ -1027,3 +1027,74 @@ def test_negative_create_bookmark_with_blank_query(self):
...
+
+    @stubbed()
+    @tier3
+    def test_mails_for_newtimezone(self):
+        """@Test: Check if mails are received to new timezone set by user
+
+        @Feature: User timezone
+
+        @Steps:
+        1.Change the timezones for user in Administer -> Users tab
```
Comment (elyezer) [inline tests/foreman/ui/test_user.py]: Add a blank line between steps tag and the steps list.

**Comment by elyezer** [comment]
I have looked over the stubs and they are good. I commented for some rst related stuff and also need to write the test names following the convention introduced by #3083.

I will ACK and let @sthirugn take a better look since he will be able to ensure the test naming convention.

**Comment by tkolhar** [comment]
@sthirugn re-named the tests according to naming conventions
please review it
thanks !!

**Comment by elyezer** [comment]
ACK

File: tests/foreman/ui/test_dashboard.py [hunk 15]
```diff
@@ -0,0 +1,197 @@
...
+        @Feature: Dashboard
+
+        @Steps:
+
+        1.Navigate to Monitor -> Content Dashboard
+        2.Try to remove some widgets
+        3.Select the Manage Dropdown box
+        4.Save the Dashboard
+
+        @Assert: Dashboard Widgets are saved successfully.
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: Dashboard is saved successfully and the removed widgets does not appear.

File: tests/foreman/ui/test_dashboard.py [hunk 16]
```diff
@@ -0,0 +1,197 @@
...
+
+        @Feature: Dashboard
+
+        @Steps:
+
+        1.Navigate to Monitor -> Content Dashboard
+        2.Try to remove some widgets
+        3.Select the Manage Dropdown box
+        4.Save the Dashboard
+        5.Dashboard Widgets are saved successfully
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: Dashboard is saved successfully?

File: tests/foreman/ui/test_dashboard.py [hunk 17]
```diff
@@ -0,0 +1,197 @@
...
+        6.Click Reset to default
+
+        @Assert: Dashboard is reset to default.
+
+        @Status: Manual
+        """
+
+    @stubbed()
+    @tier3
+    def test_positive_addwidgets(self):
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: test_positive_add_widgets

File: tests/foreman/ui/test_dashboard.py [hunk 18]
```diff
@@ -0,0 +1,197 @@
...
+        3.Add Widgets
+
+        @Assert: User is able to add widgets.
+
+        @Status: Manual
+        """
+
+    @stubbed()
+    @tier3
+    def test_positive_bookmark_search(self):
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: test_positive_search_bookmark

File: tests/foreman/ui/test_dashboard.py [hunk 19]
```diff
@@ -0,0 +1,197 @@
...
+    def test_positive_host_configuration_status(self):
+        """@Test: Check if Host Configuration Status
+        Widget links are working
+
+        @Feature: Dashboard
+
+        @Steps:
+
+        1.Navigate to Monitor -> Content Dashboard
+        2.Focus on the Host Configuration Status
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: Review the Host Configuration Status widget

File: tests/foreman/ui/test_dashboard.py [hunk 20]
```diff
@@ -0,0 +1,197 @@
...
+        3.Select each of the link which
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: Navigate to each of the links which has search string associated with it.

File: tests/foreman/ui/test_dashboard.py [hunk 21]
```diff
@@ -0,0 +1,197 @@
...
+        has search string associated with it
+
+        @Assert: Make sure each search takes you to correct info
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: @Assert: Each link shows the right info

File: tests/foreman/ui/test_dashboard.py [hunk 22]
```diff
@@ -0,0 +1,197 @@
...
+
+        @Status: Manual
+        """
+
+    @stubbed()
+    @tier3
+    def test_positive_host_configuration(self):
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: test_positive_host_configuration_chart?

File: tests/foreman/ui/test_dashboard.py [hunk 23]
```diff
@@ -0,0 +1,197 @@
...
+        @Feature: Dashboard
+
+        @Steps:
+
+        1.Navigate to Monitor -> Content Dashboard
+        2.Focus on Host Configuration Chart
+        3.Select the Circular chart which
+        has search string associated with it
+
+        @Assert: Make sure search string takes you to correct info
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: @Assert: Each link shows the right info

File: tests/foreman/ui/test_dashboard.py [hunk 24]
```diff
@@ -0,0 +1,197 @@
...
+        """@Test: Check if Host Configuration Chart
+        is working in Dashboard UI
+
+        @Feature: Dashboard
+
+        @Steps:
+
+        1.Navigate to Monitor -> Content Dashboard
+        2.Focus on Host Configuration Chart
+        3.Select the Circular chart which
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: Navigate to each of the links which has search string associated with it.

File: tests/foreman/ui/test_dashboard.py [hunk 25]
```diff
@@ -0,0 +1,197 @@
...
+
+        @Assert: Make sure search string takes you to correct info
+
+        @Status: Manual
+        """
+
+    @stubbed()
+    @tier3
+    def test_positive_task_status(self):
+        """@Test: Check if Task Status Dashboard UI
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: Test: Check if Task Status Dashboard UI is displayed

File: tests/foreman/ui/test_dashboard.py [hunk 26]
```diff
@@ -0,0 +1,197 @@
...
+
+        @Feature: Dashboard
+
+        @Steps:
+
+        1.Navigate to Monitor -> Content Dashboard
+        2.Focus on Task Status
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: Review Task Status widget

File: tests/foreman/ui/test_dashboard.py [hunk 27]
```diff
@@ -0,0 +1,197 @@
...
+    def test_positive_host_configuration(self):
+        """@Test: Check if Host Configuration Chart
+        is working in Dashboard UI
+
+        @Feature: Dashboard
+
+        @Steps:
+
+        1.Navigate to Monitor -> Content Dashboard
+        2.Focus on Host Configuration Chart
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: Review the Host Configurat Chart widget

File: tests/foreman/ui/test_dashboard.py [hunk 28]
```diff
@@ -0,0 +1,197 @@
...
+    def test_positive_warning_error_task_status(self):
+        """@Test: Check if Latest Warning/Error
+        Tasks Status are working in Dashboard UI
+
+        @Feature: Dashboard
+
+        @Steps:
+
+        1.Navigate to Monitor -> Content Dashboard
+        2.Focus on Latest Warning/Error Tasks
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: Review the Latest Warning/Error Tasks widget

File: tests/foreman/ui/test_dashboard.py [hunk 29]
```diff
@@ -0,0 +1,197 @@
...
+    def test_positive_contentview_history(self):
+        """@Test: Check if Content View History
+           are working in Dashboard UI
+
+        @Feature: Dashboard
+
+        @Steps:
+
+        1.Navigate to Monitor -> Content Dashboard
+        2.Focus on Content View History
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: Review the Content View History widget

File: tests/foreman/ui/test_dashboard.py [hunk 30]
```diff
@@ -0,0 +1,197 @@
...
+        (the environment to which it is published)
+
+        @Status: Manual
+        """
+
+    @stubbed()
+    @tier3
+    def test_positive_discovered_host(self):
+        """@Test: Check if user can access
+           Discovered Host [Widget]
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: Discovered Host widget

File: tests/foreman/ui/test_dashboard.py [hunk 31]
```diff
@@ -0,0 +1,197 @@
...
+
+        @Feature: Dashboard
+
+        @Steps:
+
+        1.Navigate to Monitor -> Content Dashboard
+        2.Focus on Discovered Hosts
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: Review the Discovered Hosts widget

File: tests/foreman/ui/test_dashboard.py [hunk 32]
```diff
@@ -0,0 +1,197 @@
...
+    @stubbed()
+    @tier3
+    def test_positive_save(self):
+        """@Test: Save the Dashboard UI
+
+        @Feature: Dashboard
+
+        @Steps:
+
+        1.Navigate to Monitor -> Content Dashboard
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: Replace all instances of this line to 
1. Navigate to Monitor -> Dashboard

File: tests/foreman/ui/test_dashboard.py [hunk 33]
```diff
@@ -0,0 +1,197 @@
...
+        3.Click each link
+
+        @Assert: Make sure search string takes you to correct info
+
+        @Status: Manual
+        """
+
+    @stubbed()
+    @tier3
+    def test_positive_warning_error_task_status(self):
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: test_positive_latest_warning_error_tasks

File: tests/foreman/ui/test_dashboard.py [hunk 34]
```diff
@@ -0,0 +1,197 @@
...
+        """@Test: Check if Latest Warning/Error
+        Tasks Status are working in Dashboard UI
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: Tasks widget is working in Dashboard UI

File: tests/foreman/ui/test_dashboard.py [hunk 35]
```diff
@@ -0,0 +1,197 @@
...
+        2.Focus on Latest Warning/Error Tasks
+
+        @Assert: The links to all failed/warnings tasks are working
+
+        @Status: Manual
+        """
+
+    @stubbed()
+    @tier3
+    def test_positive_contentview_history(self):
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: test_positive_content_view_history

File: tests/foreman/ui/test_dashboard.py [hunk 36]
```diff
@@ -0,0 +1,197 @@
...
+        """@Test: Check if Content View History
+           are working in Dashboard UI
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: widget is working in Dashboard UI (please change this in all occurences after this)

File: tests/foreman/ui/test_dashboard.py [hunk 37]
```diff
@@ -0,0 +1,197 @@
...
+
+        @Assert: Each Content View link shows its current status
+        (the environment to which it is published)
+
+        @Status: Manual
+        """
+
+    @stubbed()
+    @tier3
+    def test_positive_discovered_host(self):
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: test_positive_discovered_hosts

File: tests/foreman/ui/test_user.py [hunk 38]
```diff
@@ -1027,3 +1027,78 @@ def test_negative_create_bookmark_with_blank_query(self):
...
+        3.The left corner displays time according to the new timezone set
+
+        @Assert: Dashboard UI displays new time based on the new timezone
+
+        @Status: Manual
+        """
+
+    @stubbed()
+    @tier3
+    def test_positive_logfiles_shows_newtime(self):
```
Comment (sthirugn) [inline tests/foreman/ui/test_user.py]: new_time

File: tests/foreman/ui/test_user.py [hunk 39]
```diff
@@ -1027,3 +1027,78 @@ def test_negative_create_bookmark_with_blank_query(self):
...
+        @Feature: User timezone
+
+        @Steps:
+
+        1.Change the timezones for user in Administer -> Users tab
+        2.Navigate to Administer -> Users tab
+        3.Make sure under Email Preferences -> Mail Enabled
+        4.Send daily/weekly/monthly mails
+
+        @Assert: Emails are send according to new timezone set
```
Comment (sthirugn) [inline tests/foreman/ui/test_user.py]: s/send/sent

**Comment by sthirugn** [comment]
I dont see test cases for these scenarios. can you please review and add?

Add test case for Latest Events widget

Add test case for Sync Overview widget

Add test case for Content Host Subscription Status widget

Add test case for Current Subscription Totals widget

Add test case for Host Collections widget

Add test case for Run distribtuion in the last 30 minutes widget

Add test case for Latest Errata widget

Add a new test case - test_positive_remove_widget. This should assert - 
 * the widget is removed
 * the removed widget should be listed under Manage -> Add Widgets

Add a new test case - test_positive_add_removed_widget

Add a new test case - test_positive_minimize_widget.  This should assert - 
 * the minimized widget should be listed under Manage -> Restore Widgets

Add a new test case - test_positive_restore_minimized_widget

Add a new test case - test_positive_restore_widget.

Add a new test case - test_toggle_auto_refresh.

Add a new test case - test_positive_search

Add a new test case - test_positive_clear_search_box

**Comment by sthirugn** [comment]
Pending Comments

File: tests/foreman/ui/test_dashboard.py (hunk 16 above)
Comment (rplevka) [inline tests/foreman/ui/test_dashboard.py]: @sthirugn the notification message reads: "Widget positions successfully saved." so perhaps that could be an appropriate description too.

Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: yeah +1 - I did not see the notification message.

**Comment by tkolhar** [comment]
@sthirugn  I have added the new test scenarios along with the fixes for
previous test cases .
please review it
thanks

File: tests/foreman/ui/test_dashboard.py [hunk 40]
```diff
@@ -0,0 +1,485 @@
...
+    @tier3
+    def test_positive_host_configuration_status(self):
+        """@Test: Review the Host Configuration Status widget
+
+        @Feature: Dashboard
+
+        @Steps:
+
+        1.Navigate to Monitor -> Dashboard
+        2.Focus on the Host Configuration Status
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: I already commented to change this to 'Review the Host Configuration Status widget'.

File: tests/foreman/ui/test_dashboard.py [hunk 41]
```diff
@@ -0,0 +1,485 @@
...
+        3.Click on the list of Discovered Hosts
+
+        @Assert: It takes you to discovered hosts
+
+        @Status: Manual
+        """
+
+    @stubbed()
+    @tier3
+    def test_positive_latest_event_widget(self):
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: Please stick to the same name as UI - this should be `test_positive_latest_events_widget`

File: tests/foreman/ui/test_dashboard.py [hunk 42]
```diff
@@ -0,0 +1,485 @@
...
+        """@Test: Check if Latest Event
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: events

File: tests/foreman/ui/test_dashboard.py [hunk 43]
```diff
@@ -0,0 +1,485 @@
...
+    @stubbed()
+    @tier3
+    def test_positive_latest_warning_error_tasks(self):
+        """@Test: Check if Latest Warning/Error
+        Tasks widget is working in Dashboard UI
+
+        @Feature: Dashboard
+
+        @Steps:
+
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: This blank line is not needed, please remove all instances of this.

File: tests/foreman/ui/test_dashboard.py [hunk 44]
```diff
@@ -0,0 +1,485 @@
...
+        2.Navigate Monitor -> Dashboard
+        3.Review the Content Host Subscription Status
+        4.Click each link :
+          a.Invalid Subscriptions
+          b.Insufficient Subscriptions
+          c.Current Subscriptions
+
+        @Assert: The widget is updated with
+        all details for Current,
+        Invalid and Insufficient Subscriptions
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: The assert can fit in two lines.  Remember each line can contain upto 79 characters.

File: tests/foreman/ui/test_dashboard.py [hunk 45]
```diff
@@ -0,0 +1,485 @@
...
+    def test_positive_current_subscription_totals(self):
+        """@Test: Check if Current Subscription Totals
+           widget is working in Dashboard UI
+
+        @Feature: Dashboard
+
+        @Steps:
+
+        1.Make sure sat6 has some active subscriptions
+        2.Navigate -> Dashboard
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: Navigate to Monitor -> Dashboard?

File: tests/foreman/ui/test_dashboard.py [hunk 46]
```diff
@@ -0,0 +1,485 @@
...
+    @stubbed()
+    @tier3
+    def test_positive_host_collections(self):
+        """@Test: Check if Host Collections
+           widget is working in Dashboard UI
+
+        @Feature: Dashboard
+
+        @Steps:
+
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: Make sure to have some hosts and host collections.

File: tests/foreman/ui/test_dashboard.py (hunk 44 above)
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: Update all references similar to this.

**Comment by sthirugn** [comment]
Please fix assert statements across all test cases to have a maximum of 79 characters per line.

**Comment by sthirugn** [comment]
Pending comments

File: tests/foreman/ui/test_dashboard.py (hunk 43 above)
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: Ignore my comment.  Any multiline section should have a blank line for proper doc rendering.

**Comment by tkolhar** [comment]
@sthirugn : please review it
thanks !!

**Comment by sthirugn** [comment]
Not all tests here are tier3 in my opinion.  Can you please revisit tier classification.

File: tests/foreman/ui/test_dashboard.py (hunk 43 above)
Comment (rplevka) [inline tests/foreman/ui/test_dashboard.py]: @sthirugn - fyi, you can delete your comment :)

File: tests/foreman/ui/test_dashboard.py [hunk 47]
```diff
@@ -0,0 +1,483 @@
...
+        5.Dashboard Widgets are saved successfully
+        6.Click Reset to default
+
+        @Assert: Widget positions successfully saved.
+
+        @Status: Manual
+        """
+
+    @stubbed()
+    @tier2
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: s/tier2/tier1

File: tests/foreman/ui/test_dashboard.py [hunk 48]
```diff
@@ -0,0 +1,483 @@
...
+        2.Select Manage Dropdown box
+        3.Add Widgets
+
+        @Assert: User is able to add widgets.
+
+        @Status: Manual
+        """
+
+    @stubbed()
+    @tier2
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: s/tier2/tier1

File: tests/foreman/ui/test_dashboard.py [hunk 49]
```diff
@@ -0,0 +1,483 @@
...
+        2.Try to remove some widget
+
+        @Assert: Widget is removed
+        The widget is listed under Manage -> Add Widget
+
+        @Status: Manual
+        """
+
+    @stubbed()
+    @tier3
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: s/tier3/tier1

File: tests/foreman/ui/test_dashboard.py [hunk 50]
```diff
@@ -0,0 +1,483 @@
...
+        4.The widget is listed under Manage -> Add Widget
+        5.Click to add the widget back
+
+        @Assert: The widget is added back to the Dashboard
+
+        @Status: Manual
+        """
+
+    @stubbed()
+    @tier2
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: s/tier2/tier1

File: tests/foreman/ui/test_dashboard.py [hunk 51]
```diff
@@ -0,0 +1,483 @@
...
+        2.Try to minimize some widget
+
+        @Assert: Widget is minimized
+        The widget is listed under Manage -> Restore Widget
+
+        @Status: Manual
+        """
+
+    @stubbed()
+    @tier3
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: s/tier3/tier1

File: tests/foreman/ui/test_dashboard.py [hunk 52]
```diff
@@ -0,0 +1,483 @@
...
+        5.Click to add the widget back
+
+        @Assert: The widget is added
+        back to the Dashboard
+
+        @Status: Manual
+        """
+
+    @stubbed()
+    @tier2
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: s/tier2/tier1

File: tests/foreman/ui/test_dashboard.py [hunk 53]
```diff
@@ -0,0 +1,483 @@
...
+        1.Navigate to Monitor -> Dashboard
+        2.Add a filter to search box (eg. environment)
+
+        @Assert: Data displayed according to search box
+
+        @Status: Manual
+        """
+
+    @stubbed()
+    @tier2
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: s/tier2/tier1

File: tests/foreman/ui/test_user.py [hunk 54]
```diff
@@ -1027,3 +1027,78 @@ def test_negative_create_bookmark_with_blank_query(self):
...
+        3.Select the Timezone Dropdown list
+        4.Try to apply some timezone
+
+        @Assert: User should be able to change timezone
+
+        @Status: Manual
+        """
+
+    @stubbed()
+    @tier2
```
Comment (sthirugn) [inline tests/foreman/ui/test_user.py]: s/tier2/tier1

File: tests/foreman/ui/test_dashboard.py [hunk 55]
```diff
@@ -0,0 +1,483 @@
...
+from robottelo.test import UITestCase
+
+
+class DashboardTestCase(UITestCase):
+    """Tests for Dashboard UI"""
+
+    @stubbed()
+    @tier1
+    def test_positive_save(self):
+        """@Test: Save the Dashboard UI
```
Comment (sthirugn) [inline tests/foreman/ui/test_dashboard.py]: please remove `@Test:`. It is not needed.  Please fix all tests.

**Comment by tkolhar** [comment]
@sthirugn please review it
thanks !!

**Comment by sthirugn** [comment]
@tkolhar I asked to remove just the prefix `@Test:.` but you removed the entire test case name.  Please update the test case name again without the prefix `@Test:`.  Can you please look at some other recent tests for example? like https://github.com/SatelliteQE/robottelo/pull/3246/files

**Comment by sthirugn** [comment]
ACK

**Pull request merged**